src/shared/find-esp.c

```diff
@@ -705,21 +701,22 @@ static int verify_xbootldr(
                 sd_id128_t *ret_uuid,
                 dev_t *ret_devid) {
 
-        bool relax_checks;
         dev_t devid = 0;
         int r;
 
         assert(p);
 
-        relax_checks =
-                getenv_bool("SYSTEMD_RELAX_XBOOTLDR_CHECKS") > 0 ||
-                detect_container() > 0;
+        /* If relaxed checks are requested, don't require the xbootldr directory to be rhe root of the
+         * filesystem so that image builders can install the bootloader to a regular directory before
+         * packing that directory up into its own partition. */
+        if (getenv_bool("SYSTEMD_RELAX_XBOOTLDR_CHECKS") > 0)
+                goto finish;
```

Comment:
Hmmm. This function is badly named: is says "verify" but it actually returns some ids. I think it should be renamed.

Now with the check those fields are always set to NULL. This feels risky…

Reply:
This was already the case since before the ids were only set if `detect_container()` return `<= 0`. This seems like something that can be addressed in a separate PR

Reply:
I know. But your PR is digging the hole deeper ;)